cmd/skywire-cli/commands/config/gen.go

```diff
@@ -149,39 +181,70 @@ var genConfigCmd = &cobra.Command{
 					logger.Info("Ignoring -f --force flag, config not found.")
 				}
 			}
-			if !regen {
-				//check if the config exists
-				if _, err := os.Stat(confPath); err == nil {
-					//error config exists !regen
-					logger.Fatal("Config file already exists. Specify the '-r --regen' flag to regenerate.")
+		}
+		// skywire-cli config gen -p
+		if !stdout && outunset {
+			if pkgEnv && (selectedOS == "linux") {
+				configName = skyenv.Configjson
+				confPath = skyenv.SkywirePath + "/" + configName
+				output = confPath
+			}
+			if usrEnv {
+				confPath = skyenv.HomePath() + "/" + skyenv.ConfigName
+				output = confPath
+			}
+		}
+		if !regen && !stdout {
+			//check if the config exists
+			if _, err := os.Stat(confPath); err == nil {
+				//error config exists !regen
+				logger.Fatal("Config file already exists. Specify the '-r --regen' flag to regenerate.")
+			}
+		}
+		//don't write file with stdout
+		if !stdout {
+			userLvl, err := user.Current()
+			if err != nil {
+				logger.WithError(err).Error("Failed to detect user.")
+			} else {
+				if userLvl.Username == "root" {
+					root = true
+				}
+			}
+			//dont write config as root to non root owned dir & vice versa
+			if _, err = exec.LookPath("stat"); err == nil {
+				confPath1, _ := filepath.Split(confPath)
+				if confPath1 == "" {
+					confPath1 = "./"
+				}
+				owner, err := script.Exec(`stat -c '%U' ` + confPath1).String()
+				if err != nil {
+					logger.Error("cannot stat: " + confPath1)
+				}
+				rootOwner, err := script.Exec(`stat -c '%U' /root`).String()
```

Comment:
I have not tested on those, i think mac will use the same name for root user (root) but it may not have /root as a directory

I will change this to only happen on linux